### PR TITLE
Document preference for capitalized log messages

### PR DESCRIPTION
**Pull request by codingllama**
Document the preferred logging style for messages (capitalized vs lowercase).

### CODE REVIEW

**Comment by codingllama** [comment]
This comes up somewhat often in review, so I think we should make a call to what is preferred. I looked for corresponding linters, but it looks like none check for this today. (This might be for the best, as we'd have loads of violations regardless of the decision.)

File: rfd/0154-logging-guidelines.md [hunk 1]
```diff
@@ -211,6 +211,8 @@ to `log/slog`, we should do our best to start following a few simple rules.
    a `context.Background` to indicate as such.
 1) The message should be a fragment, and not a full sentence. Terminating the message with punctuation should be
    avoided.
+1) Starting the message with capitalization is preferred ("Speed threshold reached" instead of "speed threshold
+   reached").
```
Comment (espadolini) [inline rfd/0154-logging-guidelines.md:215]: Ok, this one I definitely disagree - if it starts with a capital letter it ought to be a full phrase and it should have proper punctuation, IMO.

Comment (codingllama) [reply rfd/0154-logging-guidelines.md:215]: I don't mind the particular way it goes, but I would like to have _a_ standard. Capitalized seems to be what the RFD does (most of the time at least).

Comment (espadolini) [reply rfd/0154-logging-guidelines.md:215]: (I prefer the style with lowercase at the beginning of the message and no punctuation, tho)

Comment (codingllama) [reply rfd/0154-logging-guidelines.md:215]: Some quick counts show we seems to favor capitalization too:

(Warning, gross and imprecise script coming up.)

```shell
# total
$ rg 'Info(|f|Context)\(.*?".+?"' --type go | awk -F'"' '{print $2}' | wc -l
> 1012

# Capitalized
$ rg 'Info(|f|Context)\(.*?".+?"' --type go | awk -F'"' '{print $2}' | grep -E '^[A-Z]' | wc -l
> 864

# rg is alias that is somewhat like this:
# /opt/homebrew/bin/rg -. -g=!*.pb.go -g=!*.svg -g=!.git/ -g=!vendor/* 
```

864 if the 1012 info samples are capitalized.

**Review by strideynet — APPROVED** [review]
This aligns with my preference - but I'm honestly just happy to see anything codified (whether that's upper or lower case)

**Comment by codingllama** [comment]
> Without a linter I wonder how impactful this will be to take a stance one way or another. As has already been pointed out in comments on this change, we are wildly inconsistent.

We will still be inconsistent, but at least there will be guidance to refer to. We can always pursue a linter, but I would take the guidance regardless (regardless of the direction it takes).

**Comment by codingllama** [comment]
Friendly ping @rosstimothy @zmb3 @espadolini ?

**Comment by zmb3** [comment]
I don't think this is something worth wasting cycles on.

If you don't enforce any change, then putting your preference in this RFD isn't going to have any impact.

-1 from me, for no reason other than there are better things for us to put our energy into.

**Comment by codingllama** [comment]
Closed.

**Pull request closed without merging**